### PR TITLE
Better medium defaults for 'button-color'

### PR DESCRIPTION
**Pull request by veryprofessionaldodo**
| - | - |
| --- | --- |
| Issue | Related to [#302](https://github.com/ripe-tech/ripe-pulse/issues/302) |
| Decisions | Currently, the `button-color` component, doesn't adapt well to forcing the height property in the loader and icon. This was occurring in `ripe-pulse` where manual styling was needed. This PR adds better functioning defaults for the `medium` size for the button. |
| Screenshots | image image |

### CODE REVIEW

**Review by beneditomauro — APPROVED** [review]

File: vue/components/ui/atoms/button-color/button-color.vue [hunk 1]
```diff
@@ -124,6 +124,10 @@
     border-color: #2d2d2d;
 }
 
+.button-color.button-color.button-color-medium {
```
Comment (NFSS10) [inline vue/components/ui/atoms/button-color/button-color.vue:127]: This doesn't seem right, shouldn't it be `.button-color.button-color-medium`

Comment (NFSS10) [reply vue/components/ui/atoms/button-color/button-color.vue:127]: Also this needs to go before before line 77, before `.button-color.button-color-small`

**Review by NFSS10 — CHANGES_REQUESTED** [review]

**Review by NFSS10 — APPROVED** [review]

**Comment by veryprofessionaldodo** [comment]
Pinging @joamag , just to not let this get forgotten

**Comment by veryprofessionaldodo** [comment]
@hugo-gomes Why were reviews re-requested?

**Comment by veryprofessionaldodo** [comment]
@hugo-gomes Why were reviews re-requested?

**Review by NFSS10 — APPROVED** [review]

**Review by beneditomauro — APPROVED** [review]

**Comment by joamag** [comment]
@joao-conde check this out

File: vue/components/ui/atoms/button-color/button-color.vue [hunk 2]
```diff
@@ -252,6 +261,12 @@
     width: 32px;
 }
 
+.button-color.button-color-medium ::v-deep .loader > div {
+    height: 28px;
+    margin-left: 2px;
+    width: 28px;
+}
```
Comment (joao-conde) [inline vue/components/ui/atoms/button-color/button-color.vue:268]: Same for margin-left, none of the other variants have it.

**Review by joao-conde — COMMENTED** [review]
@veryprofessionaldodo I believe most of this CSS should be on the place where you are using the button. Because by changing the component you change all other places that used it, which were working fine, right?

File: vue/components/ui/atoms/button-color/button-color.vue [hunk 3]
```diff
@@ -73,6 +73,10 @@
     vertical-align: middle;
 }
 
+.button-color.button-color-medium {
+    max-height: 34px;
+}
```
[the remark this answers is not in the document]
Comment (veryprofessionaldodo) [reply vue/components/ui/atoms/button-color/button-color.vue:78]: You're right, they have the `height` defined, changing that now.

File: vue/components/ui/atoms/button-color/button-color.vue (hunk 2 above)
Comment (veryprofessionaldodo) [reply vue/components/ui/atoms/button-color/button-color.vue:268]: It's something that could be addressed on the other side, removing it

**Comment by joao-conde** [comment]
@veryprofessionaldodo CI failing probably due to: https://github.com/ripe-tech/peri-shipping/pull/137#issuecomment-1135772027

**Comment by joao-conde** [comment]
@veryprofessionaldodo the problem with this PR is that you are changing a component with specific CSS for a specific usage of the button medium (in this case the export CSV button).

For example in RIPE Utils we use the button, and with your changes the UI is affected:

ship-master

ship-pr

Likewise, there are other places that this change affects.

Instead I propose you open a PR on RIPE Pulse with:
- the custom CSS you need to make it look like the design in the page where you use the button
- use the `"small"` variant as it looks to be a better fit I think (it aligns with the search bar on Pulse)
- while you are there, add a padding to the "Group Orders" button as well (that only shows when you tick one order)

image

**Pull request closed without merging**